web/yo/app/scripts/controllers/gene.js

```diff
@@ -1210,66 +1276,169 @@ angular.module('oncokbApp')
+                    uuids = [tumor.prevalence_uuid];
+                    break;
+                case 'PROGNOSTIC_IMPLICATION':
+                    reviewObjs = [tumor.progImp_review];
+                    uuids = [tumor.progImp_uuid];
+                    break;
+                case 'NCCN_GUIDELINES':
+                    reviewObjs = [tumor.nccn.therapy_review, tumor.nccn.disease_review, tumor.nccn.version_review, tumor.nccn.description_review];
+                    uuids = [tumor.nccn.therapy_uuid, tumor.nccn.disease_uuid, tumor.nccn.version_uuid, tumor.nccn.description_uuid];
+                    break;
+                case 'CLINICAL_TRIAL':
+                    reviewObjs = [tumor.trials_review];
+                    uuids = [tumor.trials_uuid];
+                    break;
+                }
+                _.each(reviewObjs, function(reviewObj) {
+                    // to keep it consistent with add method from the front end, we only set added to mutation/tumor/treatment
+                    if (sectionFlag) {
+                        reviewObj.set('added', true);
+                    } else {
+                        reviewObj.set('unobsoleted', true);
+                    }
+                    reviewObj.set('updatedBy', User.name);
+                    reviewObj.set('updateTime', new Date().getTime());
+                });
+
+                if (uuids.length > 0) {
+                    _.each(uuids, function(uuid) {
+                        setUUIDInMeta(uuid.getText());
+                    });
+                }
+            }
+            function setUUIDInMeta(uuid) {
+                if (!uuid) return;
+                var tempMapping = $rootScope.metaModel.createMap();
+                tempMapping.set('review', true);
+                $rootScope.geneMetaData.set(uuid, tempMapping);
             }
-            $scope.applyObsolete = function(eStatus, type, mutation, tumor, TI, treatment) {
+            $scope.applyObsolete = function(eStatus, type, mutation, tumor, ti, treatment) {
+                // we do not allow obsolete item any more, only unobsolete
                 if(eStatus.get('obsolete') === 'true') {
-                    if (!$scope.status.isDesiredGene) {
-                        eStatus.set('obsolete', 'false');
-                        return true;
-                    }
-                    var evidences = prepareInsertion(type, mutation, tumor, TI, treatment);
-                    if(_.isEmpty(evidences)) {
-                        eStatus.set('obsolete', 'false');
-                        return false;
+                    eStatus.set('obsolete', 'false');
+                    // set obsoleted items as newly added
+                    setModelForUnobsoleted(type, mutation, tumor, ti, treatment);
+                }
+            };
+            function acceptSection(type, mutation, tumor, ti, treatment) {
+                var tempUUIDs = getObsoletedUUIDs(type, mutation, tumor, ti, treatment);
```

Comment:
add if(dataFromFile) option for evidencebatch update api calls and other review related APIs.